core/variant/callable.h

```diff
@@ -73,6 +73,16 @@ class Callable {
 	void call_deferredp(const Variant **p_arguments, int p_argcount) const;
 	Variant callv(const Array &p_arguments) const;
 
+	template <typename... VarArgs>
+	void call_deferred(VarArgs... p_args) const {
+		Variant args[sizeof...(p_args) + 1] = { p_args..., Variant() }; // +1 makes sure zero sized arrays are also supported.
```

Comment:
```
./core/variant/callable.h:78:54: error: invalid use of incomplete type 'Variant'
                Variant args[sizeof...(p_args) + 1] = { p_args..., Variant() }; // +1 makes sure zero sized arrays are also supported.
                                                                   ^~~~~~~~~
./core/variant/array.h:38:7: note: forward declaration of 'Variant'
class Variant;
      ^
```

Should likely be in the `.cpp` as we probably don't want to include Variant here to avoid circular deps.

Reply:
Template functions can't be defined in cpp. I found a promising workaround though.